### PR TITLE
Marcos.cifuentes/clean migration

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
The objective of this PR is to merge the clean migration with the master branch

- Domain module implementation
- Data module implementation
- Dependency Injection (with Koin) implementation
- App module refactor accordingly with the clean structure
- Minor refactor on tests

Image of the tests running:

tests passed

GIF of the app working:

app working clean

### CODE REVIEW

**Review by mauroLonghi — APPROVED** [review]

File: data/src/main/java/com/globant/data/mapper/CharacterMapper.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        Character(
+            characterResponse.id,
+            characterResponse.name,
+            getDescription(characterResponse.description),
+            transformToThumbnail(characterResponse.thumbnail)
+        )
+
+
+    private fun getDescription(description: String): String {
+        return if (description.isEmpty())
```
Comment (SartiMau) [inline data/src/main/java/com/globant/data/mapper/CharacterMapper.kt]: please add brackets ({ }) to indent the code in the if and else statement

Comment (mauroLonghi) [reply data/src/main/java/com/globant/data/mapper/CharacterMapper.kt]: Opa, yes use please kotlin way for if-result

**Review by SartiMau — CHANGES_REQUESTED** [review]
approved after comments

**Review by SartiMau — APPROVED** [review]

File: data/src/main/java/com/globant/data/mapper/CharacterMapper.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply data/src/main/java/com/globant/data/mapper/CharacterMapper.kt]: Done, thanks guys

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**